crates/ruff_linter/src/rules/pylint/rules/bad_super_call.rs

```diff
@@ -0,0 +1,130 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{
+    Expr, ExprAttribute, ExprCall, ExprName, Stmt, StmtClassDef, StmtExpr, StmtFunctionDef,
+};
+
+use crate::checkers::ast::Checker;
+
+/// ### What it does
+/// Checks to see another argument other than the current class is given as the first argument of the super builtin
+///
+/// ### Why is this bad?
+/// In Python 2.7, `super()` has to be called with its own class and `self`
+/// as arguments `(super(Cat, self))`, which can lead to a mix up of parent and child class in the code.
+///
+/// In Python 3 the recommended way is to call `super()` without arguments
+/// (see also [`super-with-arguments`](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/super-with-arguments.html)).
+///
+/// ## Problematic code
+/// ```python
+/// class Animal:
+///     pass
+///
+///
+/// class Tree:
+///     pass
+///
+///
+/// class Cat(Animal):
+///     def __init__(self):
+///         super(Tree, self).__init__()  # [bad-super-call]
+///         super(Animal, self).__init__()
+/// ```
+///
+/// ## Correct code
+/// ```python
+/// class Animal:
+///     pass
+///    
+///  
+/// class Tree:
+///     pass
+///    
+///    
+/// class Cat(Animal):
+///     def __init__(self):
+///         super(Animal, self).__init__()
+/// ```
+///
+/// ## References
+/// - [Python documentation: `class super`](https://docs.python.org/3/library/functions.html#super)
+#[violation]
+pub struct BadSuperCall {
+    bad_super_arg: String,
+}
+
+impl Violation for BadSuperCall {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let Self { bad_super_arg } = self;
+        format!("Bad first argument '{bad_super_arg}' given to super()")
+    }
+}
+
+fn is_bad_super_call<'a>(first_arg: &'a Expr, current_class_name: &str) -> Option<&'a str> {
+    let Expr::Name(ExprName { id, .. }) = first_arg else {
+        return None;
+    };
+    (id != current_class_name).then_some(id)
+}
+
+fn check_expr(expr: &Expr, current_class_name: &str) -> Option<Diagnostic> {
+    if let Expr::Call(ExprCall {
+        range,
+        func,
+        arguments,
+    }) = expr
+    {
+        if let Expr::Name(ExprName { id, .. }) = func.as_ref() {
+            if id == "super" {
```

Comment:
This seems to be valid python code:

```python
class Test:
...     def __init__():
...         super = test
...         
... 
```

That means, we have to test if `super` points to the builtin `super` call and not a local binding.

Reply:
Is there a canonical test for that?